src/notifications/container.js

```diff
@@ -0,0 +1,239 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import Waypoint from 'react-waypoint'
+
+import { LoadingIndicator, ToggleSwitch } from 'src/common-ui/components'
+import * as actions from './actions'
+import * as selectors from './selectors'
+import NotificationList from './components/NotificationList'
+import Notification from './components/Notification'
+import StatusHeading from './components/StatusHeading'
+import ReadHeader from './components/ReadHeader'
+import Button from './components/Button'
+import ActionButton from './components/ActionButton'
+import { actionRegistry } from './registry'
+import OptIn from './components/OptIn'
+
+class NotificationContainer extends Component {
+    static propTypes = {
+        unreadNotificationList: PropTypes.arrayOf(PropTypes.object).isRequired,
+        readNotificationList: PropTypes.object.isRequired,
+        init: PropTypes.func.isRequired,
+        showMoreIndex: PropTypes.string,
+        handleToggleShowMore: PropTypes.func.isRequired,
+        handleTick: PropTypes.func.isRequired,
+        onBottomReached: PropTypes.func.isRequired,
+        isLoading: PropTypes.bool.isRequired,
+        needsWaypoint: PropTypes.bool.isRequired,
+        isReadExpanded: PropTypes.bool.isRequired,
+        toggleReadExpand: PropTypes.func.isRequired,
+        isReadShow: PropTypes.bool.isRequired,
+        messageCharLimit: PropTypes.number.isRequired,
+        shouldTrack: PropTypes.bool.isRequired,
+        setShouldTrack: PropTypes.func.isRequired,
+    }
+
+    static defaultProps = {
+        messageCharLimit: 150,
+    }
+
+    componentDidMount() {
+        this.props.init()
+    }
+
+    isNotificationTruncated(message) {
+        return message.length > this.props.messageCharLimit
+    }
+
+    truncateText(message) {
+
+        const lastSpaceBeforeCutoff = message.lastIndexOf(
+            ' ',
+            this.props.messageCharLimit,
+        )
+        const trunctatedText =
+            message.substr(0, lastSpaceBeforeCutoff) + '&#8230;'
```

Comment:
You should probably use the literal unicode `…` character here so it's more obvious what's going on.